File: 模拟/14.-最长公共前缀.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        min_l = min([len(s) for s in strs])
        res = ""
        if len(strs) == 1:
            return strs[0]
        
        for i in range(min_l):
            last = strs[0][i]
            for j in range(1, len(strs)):
                if strs[j][i] == last:
                    last = strs[j][i]
                    if j == len(strs) - 1:
                        res += last
                else :
                    return res

        return res

if __name__ == "__main__":
    print("hello")
    s = Solution()
    print(s.longestCommonPrefix(["aa", "bb", "cc"]))

File: 模拟/14.最长公共前缀.cpp
#include<iostream>
#include<string>
#include<vector>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res = "";
        
        int left = 0;
        while (true) {
            string alphabet = "";
            for (int j = 0; j < strs.size(); j ++) {
                if (left >= strs[j].length() || strs[j] == "") {
                    return res;
                }
                if (alphabet == "") {
                    alphabet = strs[j][left];
                } else {
                    if (alphabet[0] != strs[j][left]) {
                        return res;
                    }
                }
            }
            left = left + 1;
            res.push_back(alphabet[0]);
        }

        return res;
    }
};

int main() {

    Solution s = Solution();
    vector<string> str_array;

    // string strs[] = {"flower", "flow", "flight"};
    string strs[] = {"a"};
    for (int i = 0; i < 3; i++) {
        str_array.push_back(strs[i]);
    }
    
    // for (int i = 0; i < str_array.size(); i++) {
    //     cout<<str_array[i]<<endl;
    // }
    string res = s.longestCommonPrefix(str_array);
    cout<<res<<endl;
    // cout<<str_array<<endl;
    return 0;

}